### ISSUE TITLE
Panic on saving struct with pointer to anonymous struct

### ISSUE THREAD

**Issue by nightlord189** [guregu/dynamo#233]
In 1.22.3 if you try to save struct with nil pointer to anonymous struct you got error "panic: reflect: call of reflect.Value.Field on zero Value":

```
package main

import (
	"fmt"
	"github.com/aws/aws-sdk-go/aws"
	"github.com/aws/aws-sdk-go/aws/session"
	"github.com/guregu/dynamo"
)

type User struct {
	ID *string `dynamo:"id,hash"`
	*Address
}

type Address struct {
	City   string
	Street string
}

func main() {
	sess := session.Must(session.NewSession())
	cli := dynamo.New(sess, &aws.Config{Region: aws.String("eu-west-1")})
	table := cli.Table("bug-save-anonymous")

	id := "1"

	user := User{
		ID:      &id,
		Address: nil,
	}
	err := table.Put(&user).Run()

	fmt.Println("save error:", err)
}
```

**Comment by guregu** [comment]
Thanks for the report, I'll take a look

**Comment by guregu** [comment]
I can get a quick fix in but curious about one detail: at the moment it will always allocate the embedded struct during decoding, so:

```go
User{Address: nil} 
// decodes to:
User{Address: new(Address)}
```

Not sure how this used to work, maybe ideally it should only allocate if any of its members exist?

**Comment by guregu** [comment]
I checked an older version before the encoding changes and it has the same behavior, so I'll release the fix now. Will investigate further.

**Comment by guregu** [comment]
Just released a fix, [v1.23.0](https://github.com/guregu/dynamo/tree/v1.23.0)

**Comment by nightlord189** [comment]
Yes, it was fixed. Thank you, @guregu!
